### ISSUE TITLE
How confirm that plugin is working?

### ISSUE THREAD

**Issue by VulcanTourist** [trakt/Plex-Trakt-Scrobbler#624]
I have an Asustor AS-6510T x64 NAS with Plex Media Server, which I interact with via Plex for Windows installed in a desktop system.  I believe that I installed the Linux version of the plugin correctly, as it appears in both Plex Media Server and Plex for Windows, and I can configure it.  I was unsure of some of the configuration settings like **Proxy - Host**.  How do I confirm that it's working as intended?  Are there log entries that confirm it?

**Comment by the-jeffski** [comment]
It writes to a log in the logs folder - in a subfolder for PMS Plugin logs  com.plexapp.plugins.trakttv.log

**Comment by VulcanTourist** [comment]
Well this can't be good... the log contains nothing but these three lines repeated ad nauseum roughly every half minute or so:

> 2022-06-05 05:35:31,030 - plex.objects.core.base           (7f52a1985b38) :  WARNING (plex.objects.core.base:156) - [Detail] Omitted attribute: offlinetranscode
2022-06-05 05:35:31,034 - plex.objects.core.base           (7f52a1985b38) :  WARNING (plex.objects.core.base:156) - [Detail] Omitted attribute: pushnotifications
2022-06-05 05:36:02,174 - plex.objects.core.base           (7f52a1985b38) :  WARNING (plex.objects.core.base:156) - [Detail] Omitted attribute: musicanalysis

**Comment by the-jeffski** [comment]
But nothing else.? You should see a load of com.plexapp.plugins.trakttv      rows so sounds like its not triggering. Maybe it needs removing and re-installing

**Comment by VulcanTourist** [comment]
Yeah... nothing else.  Perhaps I didn't install it correctly.  It was my first time messing with a Plex plugin.

I've been mulling over trying Jellyfin (data migration gives me pause), and that would render this issue moot.  OTOH, having that current data present in Trakt would mean that it's sitting there ready to be imported into Jellyfin - I'm sure there's a Trakt extension for it, too - even if I can't export directly from Plex to Jellyfin.

**Comment by the-jeffski** [comment]
Yeah, doesn't seem like its loading.  How did you install - unzip to the plugin folder?

**Comment by VulcanTourist** [comment]
> How did you install - unzip to the plugin folder?

I think that is what I did.  I later discovered the WebTools plugin and added that, but it wasn't obvious to me that I even needed it for certain, as there seemed to be history and some evolution to these plugins.

**Comment by the-jeffski** [comment]
Not sure you do if you unzip to the Trakttv.bundle folder manually.   Make sure you grab this version https://github.com/Cornishman/Plex-Trakt-Scrobbler

**Comment by VulcanTourist** [comment]
> Not sure you do if you unzip to the Trakttv.bundle folder manually. Make sure you grab this version https://github.com/Cornishman/Plex-Trakt-Scrobbler

I guess it's good that you mentioned it, because I used this one: https://github.com/trakt/Plex-Trakt-Scrobbler.

I'm going to repeat what I did with this other version.  I have no explicit instructions for this NAS (AS-6510T) because it didn't exist at the time that all this was being conceived.  (I waited for an NAS with dual 10GbE ports for SMB performance and so that we could direct-connect and skip the stupidly expensive 10GbE hub.)

**Comment by VulcanTourist** [comment]
I would share the complete log, but Github bitches at it, even when I give it a .txt extension.  What do I enter in the Proxy field?  What evidence am I looking for that it's working?  I didn't copy the Tools directory, as I didn't know where to put it or what to do with it.

**Comment by the-jeffski** [comment]
Not sure the device matters unless its stopping something.  The Plugin from Git should be downloaded as a zip and then exported here in this structure within the Plex storage locations
![image](https://user-images.githubusercontent.com/63499439/173230147-a2e20598-66f7-483f-9828-d828224034bf.png)
i.e in the Plugins folder you have a Trakttv.bundle folder then within that there is a Bundle folder.

**Comment by VulcanTourist** [comment]
That is what I have.  The log is certainly different now, none of the earlier repeated warnings.  There are lines having to do with the proxy.  I left it blank because I didn't know what belonged there.

**Comment by the-jeffski** [comment]
> That is what I have. The log is certainly different now, none of the earlier repeated warnings. There are lines having to do with the proxy. I left it blank because I didn't know what belonged there.

I think the plugin allows you to specify a web proxy if you are using one - mine is blank as its not needed.  You should see in the log folder the plugin log to confirm its running.  
![image](https://user-images.githubusercontent.com/63499439/173243131-af578486-9ffd-44f2-8db0-7ed8f75934e8.png)

**Comment by VulcanTourist** [comment]
> 2022-06-11 17:09:47,250 - requests.packages.urllib3.connectionpool (7fe49abf4b38) :  WARNING (requests.packages.urllib3.connectionpool:668) - Retrying (Retry(total=2, connect=None, read=0, redirect=None)) after connection broken by 'ProxyError('Cannot connect to proxy.')': /users/settings
2022-06-11 17:09:47,251 - requests.packages.urllib3.connectionpool (7fe49abf4b38) :  WARNING (requests.packages.urllib3.connectionpool:668) - Retrying (Retry(total=1, connect=None, read=0, redirect=None)) after connection broken by 'ProxyError('Cannot connect to proxy.')': /users/settings
2022-06-11 17:09:47,252 - requests.packages.urllib3.connectionpool (7fe49abf4b38) :  WARNING (requests.packages.urllib3.connectionpool:668) - Retrying (Retry(total=0, connect=None, read=0, redirect=None)) after connection broken by 'ProxyError('Cannot connect to proxy.')': /users/settings
2022-06-11 17:09:47,253 - plugin.modules.migrations.account (7fe49abf4b38) :  WARNING (plugin.modules.migrations.account:84) - Unable to refresh trakt account (not authenticated?)

**Comment by the-jeffski** [comment]
You must have something in the proxy field in the settings panel as its trying to use one. That field should be blank.

**Comment by VulcanTourist** [comment]
I've tried it both blank and with the IP of the NAS in it (with the example port).  It was giving me those proxy errors regardless.  I don't have a network proxy.  But there was this earlier, which might mean that it did something:

>2022-06-11 17:09:46,484 - peewee_migrate.core              (7fe49abf4b38) :  INFO (peewee_migrate.core:106) - Running migrations...
2022-06-11 17:09:46,487 - peewee_migrate.core              (7fe49abf4b38) :  INFO (peewee_migrate.core:118) - Database has 17 migrations applied:
  000_initial
  001_session
  002_rule
  003_plex_account
  004_scheduler
  005_sync_result
  006_scheduler_job
  007_plex_account
  008_plex_basic_credential
  009_account
  010_account
  011_message
  012_account
  013_action
  014_session
  015_action
  016_message
2022-06-11 17:09:46,489 - peewee_migrate.core              (7fe49abf4b38) :  INFO (peewee_migrate.core:129) - Nothing to migrate
2022-06-11 17:09:46,490 - plugin.modules.migrations.main   (7fe49abf4b38) :  INFO (plugin.modules.migrations.main:38) - Running migration: 'AccountMigration'
2022-06-11 17:09:46,495 - plugin.modules.migrations.account (7fe49abf4b38) :  INFO (plugin.modules.migrations.account:244) - Plex Token: environment
2022-06-11 17:09:47,227 - plugin.models.m_plex.account     (7fe49abf4b38) :  INFO (plugin.models.m_plex.account:118) - Refreshing plex account: <PlexAccount username: None>
2022-06-11 17:09:47,235 - plugin.modules.migrations.account (7fe49abf4b38) :  WARNING (plugin.modules.migrations.account:71) - Unable to refresh plex account (not authenticated?)

**Comment by VulcanTourist** [comment]
How does that authentication work?  Do I have to visit the Trakt URL and update it every single time that I click into the settings of the plugin, or what?

**Comment by the-jeffski** [comment]
No, once you go into the settings via plex, get the pin and authenticate it will keep it and refresh your access in the background

**Comment by VulcanTourist** [comment]
> No, once you go into the settings via plex, get the pin and authenticate it will keep it and refresh your access in the background

Did I ruin things by generating a new one several times?  The process is not well documented and confusing, not least because the Trakt page makes a point of telling you it's **one use only**.

Can I even force some action from the plugin to get a clue whether it's working and how well?  It's not migrating the watched details for the series that we've been watching.  I guessed that the Tools directory had something to do with that, but it's been left behind because there were no instructions what to do with it.  I'm not a CLI expert with Linux or ADM on this NAS, but I have ShellInABox at least.

**Comment by the-jeffski** [comment]
All the pin does is fetch a new access token so redoing it won't hurt things.     

What appears in the log file once you follow that step. My only other thought is a permission issue on the files which stops it storing the new token.
